Tests/SystemTests/NetworkConnectivityTests/Subscriber/Helper/SubscriberMonitor.swift

```diff
@@ -0,0 +1,382 @@
+                // no expected location set - skip assertion
+                logHandler.debug(message: "(SKIP) expectedLocationUpdate = null", error: nil)
+                return
+            }
+
+            logHandler.debug(message: "(WAITING) expectedLocationUpdate = \(expectedLocation)", error: nil)
+
+            let lastPublishedLocation = try listenForExpectedLocationUpdate(waiterContext: waiterContext)
+
+            // TODO this seems a bit redundant – listenForExpectedLocationUpdate will either throw or return expectedLocation, won't it?
+            if lastPublishedLocation != expectedLocation {
+                logHandler.logMessage(level: .error, message: "(FAIL) lastPublishedLocation = \(lastPublishedLocation)", error: nil)
+                throw createGenericAssertionError(message: "Expected location update \(expectedLocation) but last was \(lastPublishedLocation)")
+            } else {
+                logHandler.debug(message: "(PASS) lastPublishedLocation = \(lastPublishedLocation)", error: nil)
+            }
+        }
+
+        /**
+         * Use the subscriber location flow to listen for a location update matching the one we're expecting.
+         *
+         * These location updates may arrive at any time after the trackable transitions to online, so we therefore
+         * cannot rely on the first thing we find being the "newest" state and therefore must wait for a bit.
+         */
+        private func listenForExpectedLocationUpdate(waiterContext: Blocking.WaiterContext) throws -> Location {
+            let publisher = sharedState.combineSubscriberDelegate.locations.filter { [expectedLocation] in $0.location == expectedLocation }.map(\.location)
+
+            let lastLocation = try first(publisher, label: "Wait for location \(String(describing: expectedLocation)) from subscriber", waiterContext: waiterContext)
+            logHandler.debug(message: "lastLocation: \(lastLocation)", error: nil)
+
+            return lastLocation
+        }
+
+        /**
+         * Assert that we receive the expected subscriber resolution.
+         */
+        private func assertSubscriberPreferredResolution(waiterContext: Blocking.WaiterContext) throws {
+            guard let expectedSubscriberResolution else {
+                logHandler.debug(message: "(SKIPPED) expectedSubscriberResolution = nil", error: nil)
+                return
+            }
+
+            logHandler.debug(message: "(WAITING) preferredSubscriberResolution = \(expectedSubscriberResolution)", error: nil)
+
+            let publisher = subscriberResolutionPreferences.filter { $0 == expectedSubscriberResolution }
+            let preferredSubscriberResolution = try first(publisher, label: "Wait for subscriber resolution preference \(expectedSubscriberResolution)", waiterContext: waiterContext)
+
+            // TODO this seems a bit redundant – the call to `first` will either throw or return expectedSubscriberResolution, won't it?
```

Comment:
@AndyTWF I was wondering if you had any thoughts on this — did I miss something?

Reply:
As per above :)